async/kafka/component.go

```diff
@@ -0,0 +1,92 @@
+package kafka
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/beatlabs/patron/async"
+	"github.com/beatlabs/patron/encoding"
+	"github.com/beatlabs/patron/log"
+	"strings"
+)
+
+// consumerFactoryBuilder contains all the necessary information
+// to be used for processing messages from kafka
+type consumerFactoryBuilder struct {
```

Comment:
There are already other tickets for optimizing the usage here like #70  and #54.
We should just make the proposed change and not introduce this work here because of the above.